scalyr_agent/builtin_monitors/docker_monitor.py

```diff
@@ -1624,7 +1616,56 @@ def __gather_metrics_from_api( self, containers ):
         for cid, info in containers.iteritems():
             self.__gather_metrics_from_api_for_container( info['name'] )
 
+    def get_user_agent_fragment(self):
+        """This method is periodically invoked by a separate (MonitorsManager) thread and must be thread safe."""
+
+        def _version_to_fragment(ver):
+            # Helper function to transform docker version to user-agent fragment
+            if not ver:
+                # version not yet set: return 'docker=yes' to signify docker
+                return 'docker=true'
+            log_mode = self._config.get('log_mode')
+            if log_mode == 'syslog':
+                extra = ''
+            else:
+                extra = '|raw' if self._config.get('docker_raw_logs') else '|api'
+            return 'docker=%s|%s%s' % (ver, log_mode, extra)
```

Comment:
The rationale for pipe `|` as a separator is that this character is unlikely to be part of any text returned by docker version, k8s version and any other system user-agent fragments.  This would facilitate common text processing across user-agent strings (first split by semicolon, then split by key=value, then split value on pipe).